### PR TITLE
feat: update records on changes

### PR DESCRIPTION
**Pull request by smomin**
- update algolia record when a selected field is updated
- update algolia record when no fields are configured

### CODE REVIEW

File: functions/src/util.ts [hunk 1]
```diff
@@ -40,3 +43,30 @@ export const getObjectSizeInBytes = (object: [] | {}) => {
   const recordBuffer = Buffer.from(JSON.stringify(object));
   return recordBuffer.byteLength;
 }
+
+
+export const getFields = (config) => config.fields ? config.fields.split(',') : [];
```
Comment (Haroenv) [inline functions/src/util.ts:48]: since config is also imported, should this still be an argument?

File: functions/src/util.ts [hunk 2]
```diff
@@ -40,3 +43,30 @@ export const getObjectSizeInBytes = (object: [] | {}) => {
...
+
+export const areFieldsUpdated = (config, before: DocumentSnapshot, after: DocumentSnapshot) => {
```
Comment (Haroenv) [inline functions/src/util.ts:50]: same question here

**Review by Haroenv — COMMENTED** [review]
looks reasonable!

File: functions/src/util.ts (hunk 1 above)
Comment (smomin) [reply functions/src/util.ts:48]: I debated this.  Since it was a Util function, I thought it made sense to pass in the config object.

**Review by Haroenv — APPROVED** [review]
looks correct

File: functions/lib/util.js [hunk 3]
```diff
@@ -20,3 +22,25 @@ exports.getObjectSizeInBytes = (object) => {
     const recordBuffer = Buffer.from(JSON.stringify(object));
     return recordBuffer.byteLength;
 };
+exports.getFields = (config) => config.fields ? config.fields.split(',') : [];
+exports.areFieldsUpdated = (config, before, after) => {
+    const fields = exports.getFields(config);
+    logs.debug(`fields: ${fields}`);
+    // If fields are not configured, then execute update record.
+    if (fields.lenth == 0) {
```
Comment (PeterCardenas) [inline functions/lib/util.js:30]: i think this typo will cause this function to always return false

Comment (smomin) [reply functions/lib/util.js:30]: Thanks Peter Cardenas for catching that.  Hook'em!

Comment (PeterCardenas) [reply functions/lib/util.js:30]: No problem! so cool to see another longhorn

**Pull request merged**